### DISCUSSION TITLE
Reified table constraints

### DISCUSSION THREAD

**Discussion by dariusf** [google/or-tools#3373] [Feature requests]
**What language and solver does this apply to?**
Python / CP-SAT

**Describe the problem you are trying to solve.**
I am trying to model music. I have a reified variable which is true if a note and chord are simultaneous, in which case they are constrained according to a table of allowed values.

**Describe the solution you'd like**
It would be nice if the following were supported. It is not a valid model in ortools 9.3.10497.

```python
from ortools.sat.python import cp_model

model = cp_model.CpModel()

x = model.NewIntVar(0, 3, 'x')
y = model.NewIntVar(0, 3, 'y')
b = model.NewBoolVar('b')

model.AddAllowedAssignments([x, y], [[1, 2], [2, 3]]).OnlyEnforceIf(b)

solver = cp_model.CpSolver()

solver.parameters.log_search_progress = True
status = solver.Solve(model)

print(status == cp_model.MODEL_INVALID)
# => True
```

**Comment by lperron** [comment]
will not happen anytime soon :-)
